feat: added skani species labelling and new species warnings for ANI <95%

DennisSchmitz opened the pull request:
fix: for some processes log files were not generated, now they are (see #48)
fix: made subsample.py more robust to errors (see #49)
chore: fixed some typos and minor refactors

Sofie-Hofman commented:
Looks good! I think adding comments to the iLES files is a smart idea to prevent any problems in the future

Sofie-Hofman reviewed (approved):
Looks good! I think adding comments to the iLES files is a smart idea to prevent any problems in the future

DennisSchmitz commented:
**Summary**
This update improves species identification
- Due to differing validation procedures, some groups will remain on Kraken/Bracken pending validation.
- Others will begin using Skani with this update.
- The long-term goal is for all groups to switch to Skani. Until then, both tools and their databases will be maintained.

**Changes**
- feat: Added Skani species identification with ILES compatible format (approved by stakeholders)
- fix: Ensured log files are generated for all processes (see #48)
- fix: Made `subsample.py` more robust to errors (see #49)
- chore: Fixed typos and performed minor refactors
- chore: Added comment documenting the agreed-upon ILES format (do not change without approval)
